### PR TITLE
FBXLoader2 Bug Fixes

### PR DESCRIPTION
**Pull request by Kyle-Larson**
A small couple of fixes that we encountered while using the new `FBXLoader2.js` file.
- Fixed bug with infinite looping when a vertex had more than 4 influence weights.
- Fixed a bug where not having curves defined at each axis for each transform type for an animation keyframe would not generate any keyframe.

### CODE REVIEW

File: examples/js/loaders/FBXLoader2.js [hunk 1]
```diff
@@ -571,11 +571,10 @@
 
 							var faceVertexBuffer = [];
 							var polygonIndex = 0;
-							for ( var polygonVertexIndex = 0; polygonVertexIndex < indexBuffer.length; ++ polygonVertexIndex ) {
+							indexBuffer.forEach( function ( vertexIndex, polygonVertexIndex, indexBuffer ) {
```
Comment (takahirox) [inline examples/js/loaders/FBXLoader2.js]: In the Mr.doob's Code Style...

> Don't use foreach loops.

https://github.com/mrdoob/three.js/wiki/Mr.doob%27s-Code-Style%E2%84%A2#cycles

Comment (Kyle-Larson) [inline examples/js/loaders/FBXLoader2.js]: I'm all for following code styles to keep things consistent, but can I ask why this is a rule?  It's supported by every major browser, and it helps keep temporary loop variables contained within the loop.  I could honestly just write a function and then call that function inside the loop (which isn't against his coding style), but at that point what would we be gaining?

Comment (takahirox) [inline examples/js/loaders/FBXLoader2.js]: The following is just my speculation.
We need to ask @mrdoob for the real reason.

Yeah `forEach` has a good scope but `for` is faster than it in general.
Three.js (3DCG engine) often handles long array,
so only using `for` isn't bad I think.

    function init( array ) {
        for ( var i = 0; i < 0xffffff; i ++ ) {
		array[ i ] = 1;
        }
    }

    function run1( array ) {
        var sum = 0;
        for ( var i = 0, il = array.length; i < il; i ++ ) {
            sum += array[ i ];
        }
        return sum;
    }

    function run2( array ) {
        var sum = 0;
        array.forEach( function( value ) {
            sum += value;
        } );
        return sum;
    }

    function test(func, array) {
        var startTime = performance.now();
        func(array);
        var endTime = performance.now();
        console.log(endTime - startTime);
    }

    var array = [];
    init(array);
    test(run1, array); // warm up
    test(run1, array);
    test(run2, array); // warm up
    test(run2, array);

    // result
    VM73:48 18.36999999999989
    VM73:48 32.095000000001164
    VM73:48 504.89999999999964
    VM73:48 507.89999999999964

Reference: https://jsperf.com/for-vs-foreach/9

Comment (Kyle-Larson) [inline examples/js/loaders/FBXLoader2.js]: Coming from a C++ background, this completely baffles me.  But performance is performance, so I'll go ahead and remove the for-each's

Comment (mrdoob) [inline examples/js/loaders/FBXLoader2.js]: > The following is just my speculation.
We need to ask @mrdoob for the real reason.

Yep! It's for performance reasons.

**Comment by mrdoob** [comment]
Many thanks!

**Pull request merged**